app/controllers/application_controller.rb

```diff
@@ -14,6 +14,13 @@ class ApplicationController < ActionController::Base
              forbidden: {status:403, message: "Forbbiden, must supply valid user credentials"}}
 
 
+  def company_validation
+    campus = Campus.find(params[:campus_id]) unless params[:campus_id].nil?
+    campus = Campus.find(params[:id]) if campus.nil? && params[:id].present?
+    return if current_user.nil? || campus.nil?
+    raise ForbiddenAuth unless current_user.company == campus.company
```

Comment:
Shouldn't we redirect the user with an error message instead of raising a 500 error in our app? That seems like a really jarring user experience.

I also think these if statements could be cleaned up a bit, we've got a lot going on.

line 20 should probably be above 18 and 19.

Lines 18 and 19 could probably be combined as well. With something like (but not exactly):
`campus = Campus.find_by(id: (params[:campus_id] || params[:id]))`
Then we have either nil or a campus, with no "Couldn't find company with id of nil" errors.